python-sdk/nuscenes/eval/lidarseg/render.py

```diff
@@ -0,0 +1,327 @@
+import argparse
+import json
+import os
+from typing import Dict, List, Tuple
+
+from matplotlib import pyplot as plt
+from matplotlib import gridspec
+import numpy as np
+from tqdm import tqdm
+
+from nuscenes import NuScenes
+from nuscenes.eval.lidarseg.evaluate import LidarSegEval
+from nuscenes.eval.lidarseg.utils import ConfusionMatrix, LidarsegClassMapper, load_bin_file, LidarSegPointCloud
+from nuscenes.lidarseg.lidarseg_utils import colormap_to_colors, create_lidarseg_legend, get_labels_in_coloring
+
+
+class LidarSegEvalStratified(LidarSegEval):
+    """
+    Extends the LidarSegEval class to provide an evaluation which is stratified by radial distance from the ego lidar.
+    """
+    def __init__(self, nusc: NuScenes,
+                 results_folder: str,
+                 output_dir: str,
+                 eval_set: str,
+                 strata_list: Tuple[Tuple[float]] = ((0, 20), (20, 40), (40, None)),
+                 verbose: bool = True):
+        """
+        :param nusc: A NuScenes object.
+        :param results_folder: Path to the folder where the results are stored.
+        :param output_dir: Folder to save plots and results to.
+        :param eval_set: The dataset split to evaluate on, e.g. train, val or test.
+        :param strata_list: The stratum to evaluate by, in meters.
+        :param verbose: Whether to print messages during the evaluation.
+        """
+        super().__init__(nusc, results_folder, eval_set, verbose)
+
+        self.output_dir = output_dir
+        assert os.path.exists(self.output_dir), 'Error: {} does not exist.'.format(self.output_dir)
+
+        self.strata_list = strata_list
+        # Get the display names for each strata (e.g. (40, 60) -> '40 to 60'; if the upper bound of a strata is
+        # None (e.g. (40, None)), then the display name will be '40+'.
+        self.strata_names = ['{}m to {}m'.format(rng[0], rng[1]) if rng[1] is not None else str(rng[0]) + 'm+'
+                             for i, rng in enumerate(self.strata_list)]
+
+        self.ignore_name = self.mapper.ignore_class['name']
+
+        # Create a list of confusion matrices, one for each strata.
+        self.global_cm = [ConfusionMatrix(self.num_classes, self.ignore_idx) for i in range(len(strata_list))]
+
+        # After running the evaluation, a list of dictionaries where each entry corresponds to each strata and is of
+        # the following format:
+        # [
+        #     {
+        #         "iou_per_class": {
+        #             "ignore": NaN,
```

Comment:
Can we remove ignore?

Reply:
The dict is meant to be raw (i.e. every single class, including the `ignore` class, is contained within) so that users have access to all the information if they want to take the raw metrics and customize some other visualizations around them (note that the `ignore` class is **not** displayed in our renders though)